### ISSUE TITLE
Spacebar multitaps and line break

### ISSUE THREAD

**Issue by sphh** [dessalines/thumb-key#881]
### Steps to reproduce

1. Open a text input with a proportional font.
2. Fill a line with enough characters, so that a `,` or `.` would still fit into the line, but a `?` would break the line
3. Tap the spacebar multiple times: When the line breaks, e.g. encountering the question mark, more taps on the spacebar does not select the next punctuation, but inserts a space.

### Expected behavior

More taps should cycle through the punctuations.

### Actual behavior

After the line breaks into the next line, more taps on the spacebar starts the multitap behaviour again.

### version of the program

2.7.6

### Android version

Android 8.1

### Device

Unihertz Jelly

### Acknowledgements

- [X] I have written a short but informative title.
- [X] I have updated the app to **[the latest version](https://github.com/user/repo/releases/latest)**.
- [X] I have searched the existing issues and this is a new one, **NOT** a  duplicate or related to another open issue.
- [X] This is not a question or a discussion, in which case I should have  gone to [lemmy.ml/c/thumbkey](https://lemmy.ml/c/thumbkey)
- [X] This is a **single** bug report, in case of multiple bugs I will open  a separate issue for each one  (they can always link to each other if related)
- [ ] I have admitted that I am a clown by having checked this box, as I have not read these acknowledgements. 🤡
- [X] I have filled out all of the requested information in this form.

**Comment by dessalines** [comment]
I'm guessing it considers a newline a cursor move? https://github.com/dessalines/thumb-key/blob/main/app/src/main/java/com/dessalines/thumbkey/ui/components/keyboard/KeyboardKey.kt#L210

I'm not sure what the best way to get around that would be.

**Comment by sphh** [comment]
I have absolutely no idea, since I am neither an Android nor a Java developer. I just know, that this works as expected with The Other App …

**Comment by glm4610** [comment]
> I'm guessing it considers a newline a cursor move?

Yes, the way it is implemented, it actually compares the on-screen cursor coordinates to check if it moved

> I'm not sure what the best way to get around that would be.

I've managed to get a working refactoring which actually takes the cursor's index into account. I'll create a pull request for it soon. 

On another note, I think it would be beneficial to redesign the way ReplaceLastText actions work, as they seem quite imperative currently (as they depend on a hard coded *trim length*). This redesign could provide new interesting action possibilities, and it might fix some of the problems we have with multitaps on some apps too
